DQM/Physics/src/TopSingleLeptonDQM_miniAOD.cc

```diff
@@ -0,0 +1,789 @@
+      "N_{30}(TCHP)", 10, 0., 10.);
+  // btag discriminator for track counting high purity
+  hists_["jetBDiscPur_"] = ibooker.book1D("JetBDiscPur",
+      "Disc_{TCHP}(Jet)", 100, 0., 10.);
+  // multiplicity of btagged jets (for simple secondary vertex) with pt(L2L3)>30
+  hists_["jetMultBVtx_"] = ibooker.book1D("JetMultBVtx",
+      "N_{30}(SSVHE)", 10, 0., 10.);
+  // btag discriminator for simple secondary vertex
+  hists_["jetBDiscVtx_"] = ibooker.book1D("JetBDiscVtx",
+      "Disc_{SSVHE}(Jet)", 35, -1., 6.);
+  // multiplicity for combined secondary vertex
+  hists_["jetMultCSVtx_"] = ibooker.book1D("JetMultCSV", "N_{30}(CSV)", 10, 0., 10.);
+  // btag discriminator for combined secondary vertex
+  hists_["jetBCVtx_"] = ibooker.book1D("JetDiscCSV",
+      "Disc_{CSV}(JET)", 100, -1., 2.);
+  // pt of the 1. leading jet (uncorrected)
+  hists_["jet1PtRaw_"] = ibooker.book1D("Jet1PtRaw", "pt_{Raw}(jet1)", 60, 0., 300.);
+  // pt of the 2. leading jet (uncorrected)
+  hists_["jet2PtRaw_"] = ibooker.book1D("Jet2PtRaw", "pt_{Raw}(jet2)", 60, 0., 300.);
+  // pt of the 3. leading jet (uncorrected)
+  hists_["jet3PtRaw_"] = ibooker.book1D("Jet3PtRaw", "pt_{Raw}(jet3)", 60, 0., 300.);
+  // pt of the 4. leading jet (uncorrected)
+  hists_["jet4PtRaw_"] = ibooker.book1D("Jet4PtRaw", "pt_{Raw}(jet4)", 60, 0., 300.);
+  // selected events
+  hists_["eventLogger_"] = ibooker.book2D("EventLogger",
+      "Logged Events", 9, 0., 9., 10, 0., 10.);
+
+  // set axes titles for selected events
+  hists_["eventLogger_"]->getTH1()->SetOption("TEXT");
+  hists_["eventLogger_"]->setBinLabel(1, "Run", 1);
+  hists_["eventLogger_"]->setBinLabel(2, "Block", 1);
+  hists_["eventLogger_"]->setBinLabel(3, "Event", 1);
+  hists_["eventLogger_"]->setBinLabel(4, "pt_{L2L3}(jet1)", 1);
+  hists_["eventLogger_"]->setBinLabel(5, "pt_{L2L3}(jet2)", 1);
+  hists_["eventLogger_"]->setBinLabel(6, "pt_{L2L3}(jet3)", 1);
+  hists_["eventLogger_"]->setBinLabel(7, "pt_{L2L3}(jet4)", 1);
+  hists_["eventLogger_"]->setBinLabel(8, "M_{W}", 1);
+  hists_["eventLogger_"]->setBinLabel(9, "M_{Top}", 1);
+  hists_["eventLogger_"]->setAxisTitle("logged evts", 2);
+  return;
+}
+
+void MonitorEnsemble::fill(const edm::Event& event,
+                           const edm::EventSetup& setup) {
+  // fetch trigger event if configured such
+  edm::Handle<edm::TriggerResults> triggerTable;
+
+  if (!triggerTable_.isUninitialized()) {
+    if (!event.getByToken(triggerTable_, triggerTable)) return;
+  }
+
+  /*
+  ------------------------------------------------------------
+
+  Primary Vertex Monitoring
+
+  ------------------------------------------------------------
+  */
+  // fill monitoring plots for primary verices
+  edm::Handle<edm::View<reco::Vertex> > pvs;
+  if (!event.getByToken(pvs_, pvs)) return;
+  unsigned int pvMult = 0;
+  for (edm::View<reco::Vertex>::const_iterator pv = pvs->begin();
+       pv != pvs->end(); ++pv) {
+    if (!pvSelect_ || (*pvSelect_)(*pv)) pvMult++;
```

Comment:
@pargali - confirming this is the right logic -I would have expected if ( pvSelect_ && (*pvSelect_)(*pv)) pvMult++;